builtin/providers/aws/resource_aws_api_gateway_method_settings.go

```diff
@@ -0,0 +1,248 @@
+						},
+						"throttling_rate_limit": {
+							Type:     schema.TypeFloat,
+							Optional: true,
+						},
+						"caching_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"cache_ttl_in_seconds": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"cache_data_encrypted": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"require_authorization_for_cache_control": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"unauthorized_cache_control_header_strategy": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceAwsApiGatewayMethodSettingsRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	log.Printf("[DEBUG] Reading API Gateway Method Settings %s", d.Id())
+	input := apigateway.GetStageInput{
+		RestApiId: aws.String(d.Get("rest_api_id").(string)),
+		StageName: aws.String(d.Get("stage_name").(string)),
+	}
+	stage, err := conn.GetStage(&input)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && awsErr.Code() == "NotFoundException" {
+			log.Printf("[WARN] API Gateway Stage %s not found, removing method settings", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+	log.Printf("[DEBUG] Received API Gateway Stage: %s", stage)
+
+	methodPath := d.Get("method_path").(string)
+	settings, ok := stage.MethodSettings[methodPath]
+	if !ok {
+		log.Printf("[WARN] API Gateway Method Settings for %q not found, removing", methodPath)
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("settings.0.metrics_enabled", settings.MetricsEnabled)
+	d.Set("settings.0.logging_level", settings.LoggingLevel)
+	d.Set("settings.0.data_trace_enabled", settings.DataTraceEnabled)
+	d.Set("settings.0.throttling_burst_limit", settings.ThrottlingBurstLimit)
+	d.Set("settings.0.throttling_rate_limit", settings.ThrottlingRateLimit)
+	d.Set("settings.0.caching_enabled", settings.CachingEnabled)
+	d.Set("settings.0.cache_ttl_in_seconds", settings.CacheTtlInSeconds)
+	d.Set("settings.0.cache_data_encrypted", settings.CacheDataEncrypted)
+	d.Set("settings.0.require_authorization_for_cache_control", settings.RequireAuthorizationForCacheControl)
+	d.Set("settings.0.unauthorized_cache_control_header_strategy", settings.UnauthorizedCacheControlHeaderStrategy)
+
+	return nil
+}
+
+func resourceAwsApiGatewayMethodSettingsUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigateway
+
+	methodPath := d.Get("method_path").(string)
+	prefix := fmt.Sprintf("/%s/", methodPath)
+
+	ops := make([]*apigateway.PatchOperation, 0)
+	if d.HasChange("settings.0.metrics_enabled") {
+		ops = append(ops, &apigateway.PatchOperation{
+			Op:    aws.String("replace"),
+			Path:  aws.String(prefix + "metrics/enabled"),
+			Value: aws.String(fmt.Sprintf("%t", d.Get("settings.0.metrics_enabled").(bool))),
```

Comment:
It surprised me too when I saw that for the first time, but I guess it makes sense if they want to support patching of whole objects (marshalled JSON which is just string), not just primitive data types.

TL;DR: https://github.com/aws/aws-sdk-go/blob/master/service/apigateway/api.go#L17073-L17077